utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -371,12 +372,14 @@ Status TransactionLockMgr::AcquireWithTimeout(
       // detection.
       if (wait_ids.size() != 0) {
         if (txn->IsDeadlockDetect()) {
+          env->GetCurrentTime(&unix_time);
```

Comment:
Can we call this inside `TransactionLockMgr::IncrementWaiters` when deadlock occurs? That way, we only pay the cost of getting current time if a deadlock has actually occurred.